### DISCUSSION TITLE
Basic help with filtering data then reduce collection to a single group arc

### DISCUSSION THREAD

**Discussion by vonKristoff** [d3/d3#3959] [Q&A] [answered]
Hey there ... I have some data which has all kinds of attributes. They are being plotted on a Force Graph, so have `{ x, y }` values added to them.
With the data nodes looking more like: `[{ x, y, group, strength }]`

Q. how would I be able to create a filtered collection of group  and draw a circle (arc) around all of them?

I have all my nodes as follows
```js
const nodes = data.nodes.map((d) => ({ ...d }));
```
I define a group
```js
const works = d3.group(nodes, (d) => d.group === 'Works');
```
I draw the nodes
```js
const node = svg
			.append('g')
			.attr('stroke', '#fff')
			.attr('stroke-width', 1.5)
			.selectAll('circle')
			.data(nodes)
			.join('circle')
			.attr('r', 5)
			.attr('fill', (d) => color(d.group));
```
I want to target the works collection and do something
```js
const group = svg
			.append('g')
			.attr('dashed', '#f5f')
			.attr('stroke-width', 3)
			.selectAll('circle')
			.data(w)
			.join('circle')
			.attr('r', 45)
			.attr('fill', 'rgba(0,0,0,0.125');
```
I get a circle in the center of the page only.

The real Question is - How would I manipulate the works data so I could draw a circle around all of the points?

**Comment by Fil** [comment, marked as answer]
You could probably use [d3.packEnclose](https://d3js.org/d3-hierarchy/pack#packEnclose).

The algorithm is described [here](https://observablehq.com/@d3/d3-packenclose). It's not a _basic_ question at all :)

**Comment by vonKristoff** [comment]
Amazing! ok ok - basically i am trying to get my head around what d3 does under the hood.

If i try to use this with existing data - `const work = d3.group(nodes, (d) => d.group === 'Works');`

I thought I might do this `const enclosingCircle = d3.packEnclose(work);` but it fails. So I am unsure how to unpack a group from data the d3 way - unless I just use JS and an array filter.

Any thoughts?

**Reply by Fil** [reply]
I'd recommend that you read the documentation, see what d3.group outputs, and what type of inputs d3.packEnclose expects.

**Reply by vonKristoff** [reply]
ofc! sorry chasing cheap
Thanks for your help

**Reply by Fil** [reply]
you'll also need to pass in the value for *r* (which in your case is a constant), so the answer is probably something like:
`d3.packEnclose(work.get(true).map(({x, y}) => ({x, y, r: 5})`
